spring-kafka-docs/src/main/antora/modules/ROOT/pages/kafka/headers.adoc

```diff
@@ -179,3 +179,81 @@ MessagingMessageConverter converter() {
 
 If using Spring Boot, it will auto configure this converter bean into the auto-configured `KafkaTemplate`; otherwise you should add this converter to the template.
 
+[[multi-value-header]]
+== Support multi-value header
+
+Spring for Apache Kafka 4.0 supports multi-value header where same logical header key appears more than once in a Kafka record.
+
+By default, when multiple headers share the same name, the `HeaderMapper` treats them as a single value and serialises the collection to JSON.
```

Comment:
This sentence does not reflect reality. The message headers abstraction doesn’t have multi-value headers notion.
You probably talk about a collection header which just serialized to a single Kafka header as JSON.